### PR TITLE
[script.module.kodi65] 1.2.1+matrix.5

### PR DESCRIPTION
**Pull request by fbacher**
### Description
Matrix version of script. The original owner, Philipp Temminghof
(kodi65), is listed as retired and has not responded to several notes.
As far as I can tell his last post was in 2017. scott967 made some
updates for matrix as well as a minor new feature that he required. I
made addon.xml dependency changes and other minor stuff. I am
willing to take over ownership of this addon just so that it doesn't die.

The source is at: https://github.com/fbacher/script.module.kodi65/tree/Matrix-RC1
### Checklist:

- [X ] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [ X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [X ] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by enen92** [comment]
Hey @fbacher thanks for the submission. Have you seen my comment on the forum thread? Is changing the add-on name and Id an option for you? Usually we only accept Kodi/XBMC branded add-ons if maintained/created by team Kodi.

**Comment by fbacher** [comment]
Is the add-on name script.module.utils acceptable? I don't like it much, but it is a collection of some generally small helpers without much in common in between them. I checked leia and matrix branches and there is no collision.

I assume that I don't have to change internal module names, etc. to eliminate use of kodi or xbmc, although I plan on changing a few, where it seems to make sense.

**Comment by enen92** [comment]
Anything you like is fine. Please make sure it doesn't overlap with any existing module in pypi just to avoid confusion. Please note the module is not shown in the add-on browser/to users so the name and Id doesn't need to be anything pretty

**Comment by fbacher** [comment]
I want to confirm something. In repo-scripts I upload the files that will go into the installable package (which is what I did). There is no importing or reference to my  git repo. My repo and repo-scripts are distinct and they are unaware of each other.
Correct?

**Comment by enen92** [comment]
Yeah that should be fine, this is a repository (or at least the source of the repo). You only need to make sure the src tag in add-on.xml mentions the upstream of your add-on (your git repo and project)

File: script.module.kodi65/lib/kodi65/listitem.py [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+            props['duration(h)'] = utils.format_time(infos["duration"], "h")
+            props['duration(m)'] = utils.format_time(infos["duration"], "m")
+        for key, value in props.items():
+            listitem.setProperty(key, str(value))
+        for key, value in self.specials.items():
+            listitem.setProperty(key, str(value))
+        artwork = {k: v for k, v in self._artwork.items() if v}
+        if artwork:
+            listitem.setArt(artwork)
+        if infos:
```
Comment (enen92) [inline script.module.kodi65/lib/kodi65/listitem.py:219]: Note this is now deprecated and belongs to infotag

File: script.module.kodi65/lib/kodi65/player.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        self.stopped = False
+
+    @busy.set_busy
+    def youtube_info_by_id(self, youtube_id):
+        vid = utils.get_youtube_info(youtube_id)
+        if not vid:
+            return None, None
+        listitem = xbmcgui.ListItem(label=vid.title,
+                                    thumbnailImage=vid.thumbnail)
+        listitem.setInfo(type='video',
```
Comment (enen92) [inline script.module.kodi65/lib/kodi65/player.py:35]: Deprecated

File: script.module.kodi65/lib/kodi65/player.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                         infoLabels={"genre": vid.sourceName,
+                                     "plot": vid.description})
+        return vid.streamURL(), listitem
+
+    def wait_for_video_end(self):
+        xbmc.sleep(500)
+        while not self.stopped:
```
Comment (enen92) [inline script.module.kodi65/lib/kodi65/player.py:42]: Better check also abortRequested

File: script.module.kodi65/lib/kodi65/selectdialog.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        elif control_id == C_BUTTON_CANCEL:
+            self.close()
+
+
+def open(listitems, header, detailed=True, extrabutton=False):
+    """
+    open selectdialog, return index (-1 for closing, -2 for extra button)
+    *listitems needs to be an iterable with ListItems (array, ItemList)
+    """
+    xbmc.executebuiltin("Dialog.Close(busydialog)")
```
Comment (enen92) [inline script.module.kodi65/lib/kodi65/selectdialog.py:59]: Avoid calling this in v19++, use dialogbusynocancel

File: script.module.kodi65/lib/kodi65/utils.py [hunk 5]
```diff
@@ -0,0 +1,504 @@
...
+from kodi65 import addon
+
+
+def youtube_info_by_id(youtube_id):
+    vid = get_youtube_info(youtube_id)
+    if not vid:
+        return None, None
+    url = vid.streamURL()
+    listitem = xbmcgui.ListItem(label=vid.title,
+                                thumbnailImage=vid.thumbnail,
```
Comment (enen92) [inline script.module.kodi65/lib/kodi65/utils.py:31]: Thumbnail args were removed, please use setart

File: script.module.kodi65/lib/kodi65/utils.py [hunk 6]
```diff
@@ -0,0 +1,504 @@
...
+                                path=url)
+    listitem.setInfo(type='video',
```
Comment (enen92) [inline script.module.kodi65/lib/kodi65/utils.py:33]: Deprecated

File: script.module.kodi65/lib/kodi65/utils.py [hunk 7]
```diff
@@ -0,0 +1,504 @@
...
+    return base_age
+
+
+def get_http(url, headers=False):
+    """
+    fetches data from *url, returns it as a string
+    """
+    succeed = 0
+    if not headers:
+        headers = {'User-agent': 'Kodi/17.0 ( phil65@kodi.tv )'}
```
Comment (enen92) [inline script.module.kodi65/lib/kodi65/utils.py:290]: Email not valid anymore

File: script.module.kodi65/lib/kodi65/youtube.py [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+import urllib.request, urllib.parse, urllib.error
+import itertools
+import html
+
+from kodi65 import utils
+from kodi65 import VideoItem
+from kodi65 import ItemList
+
+YT_KEY = ''
+BASE_URL = "https://www.googleapis.com/youtube/v3/"
```
Comment (enen92) [inline script.module.kodi65/lib/kodi65/youtube.py:15]: Do you have a new api key or are you reusing Phil's?

**Review by enen92 — CHANGES_REQUESTED** [review]
See comments. Thanks for the submission

**Comment by fbacher** [comment]
Closed by accident, but since I started over on my branch, it was probably just as well. New request opened.

**Pull request closed without merging**